### PR TITLE
#184651348-Shopping-cart

### PR DESCRIPTION
**Pull request by ruberwa**
#### What does this PR do?
  * User/buyer is able to add items to their cart

#### Description of Task to be completed?
  * A User who logged in is able to add the item to the cart.
  * If the Product is already in the cart, the Quantity will be updated.
  * Total Price of Quantities added is being calculated
  * Total Amount of the whole Shopping Cart is being calculated.

#### How should this be manually tested?
  * Clone the branch
  * Go to its path in the terminal run: **npm install** for Dependencies
  * Create a .env file
  * Add the following Variables in that .env file
    - JWT_SECRET
    - USER_SECRET_KEY
    - BCRYPT_KEY
    - TOKEN_EXPIRATION
    - SENDGRID_API_KEY
    - SENDER_EMAIL
    - GOOGLE_CLIENT_ID
    - GOOGLE_CLIENT_SECRET
    - NODE_ENV
    - DB_USERNAME
    - DB_PASSWORD
    - DB_HOST
    - DB_NAME
    - RESET_URL

  *  Start the app: **npm start**
  * Go to this URL to test functionalities: http://localhost:5000/api-docs/

#### What are the relevant pivotal tracker/Trello stories?
      * 184651348

### CODE REVIEW

**Review by JulesNIYOMWUNGERI — DISMISSED** [review]
well done!

**Review by kagame8732 — DISMISSED** [review]
Well implemented!!

File: src/Database/migrations/20230331030750-modify_user_add_role_field.js
Comment (Raphaela664) [inline src/Database/migrations/20230331030750-modify_user_add_role_field.js:1]: Why was this file deleted?

Comment (ruberwa) [reply src/Database/migrations/20230331030750-modify_user_add_role_field.js:1]: It was for adding a role field in the user but that field was already there!

**Review by kagame8732 — DISMISSED** [review]
Well implementation.

**Review by JulesNIYOMWUNGERI — DISMISSED** [review]
looks good!

**Review by kagame8732 — DISMISSED** [review]
Well implemented!!

**Review by bahati10 — DISMISSED** [review]
Good To Go!

**Review by heshima18 — APPROVED** [review]

**Review by heshima18 — APPROVED** [review]

**Review by kagame8732 — APPROVED** [review]
Working well!

**Review by bstwagir — APPROVED** [review]
well implemented
